src/bridge/test-peer.c

```diff
@@ -470,6 +483,110 @@ test_reopen (TestCase *tc,
+
+  while ((sent = mock_transport_pop_channel (tc->transport, "a")) == NULL)
+    g_main_context_iteration (NULL, TRUE);
+
+  cockpit_assert_bytes_eq (sent, "OH MARMALADE", -1);
+
+  other = g_object_ref (cockpit_peer_ensure (tc->peer));
+  emit_string (tc, NULL, "{\"command\": \"close\", \"channel\": \"a\"}");
+
+  while ((control = mock_transport_pop_control (tc->transport)) == NULL)
+    g_main_context_iteration (NULL, TRUE);
+  cockpit_assert_json_eq (control, "{\"command\":\"close\",\"channel\":\"a\"}");
+  control = NULL;
+
+  g_signal_connect (other, "closed", G_CALLBACK (on_other_closed), &closed);
+  while (!closed)
+    g_main_context_iteration (NULL, TRUE);
+
+  /* Sending again reopens with count at zero */
+  emit_string (tc, NULL, "{\"command\": \"open\", \"channel\": \"a\", \"payload\": \"upper\"}");
+  emit_string (tc, "a", "Oh MarmaLade");
+
+  while ((control = mock_transport_pop_control (tc->transport)) == NULL)
+    g_main_context_iteration (NULL, TRUE);
+  cockpit_assert_json_eq (control, "{\"command\":\"ready\",\"channel\":\"a\",\"count\":0}");
+  control = NULL;
+
+  while ((sent = mock_transport_pop_channel (tc->transport, "a")) == NULL)
+    g_main_context_iteration (NULL, TRUE);
+
+  /* The fallback just echos */
+  cockpit_assert_bytes_eq (sent, "OH MARMALADE", -1);
+
+  g_free (bridge);
+  g_object_unref (other);
+}
+
+static void
+test_reopen_fail (TestCase *tc,
+                  gconstpointer unused)
+{
+  gchar *link;
+  gchar *bridge;
+  GBytes *sent;
+  JsonObject *control;
+
+  link = g_strdup_printf ("%s/mock-bridge", g_get_tmp_dir ());
+  bridge = g_strdup_printf("{ \"match\": { \"payload\": \"lower\" }, \"spawn\": [ \"%s\", \"--lower\" ], \"problem\": \"custom-problem\" }", link);
+  tc->peer = peer_new (tc->transport, bridge);
+
+  if (g_file_test (link, G_FILE_TEST_EXISTS))
+    g_assert (g_unlink (link) == 0);
+
+  emit_string (tc, NULL, "{\"command\": \"open\", \"channel\": \"a\", \"payload\": \"lower\"}");
+  emit_string (tc, "a", "Oh Marmalade");
+
+  while ((control = mock_transport_pop_control (tc->transport)) == NULL)
+    g_main_context_iteration (NULL, TRUE);
+  cockpit_assert_json_eq (control, "{\"command\":\"close\", \"channel\": \"a\", \"problem\":\"custom-problem\"}");
+  control = NULL;
+
+  cockpit_peer_reset (tc->peer);
+  g_assert (symlink (BUILDDIR "/mock-bridge", link) == 0);
+
+  /* Sending again reopens */
+  emit_string (tc, NULL, "{\"command\": \"open\", \"channel\": \"a\", \"payload\": \"lower\"}");
+  emit_string (tc, "a", "Oh MarmaLade");
+
+  while ((control = mock_transport_pop_control (tc->transport)) == NULL)
+    g_main_context_iteration (NULL, TRUE);
+  cockpit_assert_json_eq (control, "{\"command\":\"ready\",\"channel\":\"a\"}");
+  control = NULL;
+
+  while ((sent = mock_transport_pop_channel (tc->transport, "a")) == NULL)
+    g_main_context_iteration (NULL, TRUE);
+
+  /* The fallback just echos */
+  cockpit_assert_bytes_eq (sent, "oh marmalade", -1);
```

Comment:
The mock-bridge exists in this second part, and apparently you actually intend the "lower" bridge to work here as you compare to the lower-case string. Is the "The fallback just echos" comment just a copy&paste leftover here, or am I missing something?